src/wrapper/refs_wrapper.cpp

```diff
@@ -0,0 +1,17 @@
+#include "../utils/git_exception.hpp"
+#include "../wrapper/refs_wrapper.hpp"
+
+
+reference_wrapper::~reference_wrapper()
+{
+    git_reference_free(p_resource);
+    p_resource=nullptr;
+}
+
+const char* reference_wrapper::get_ref_name(const repository_wrapper& rw)
```

Comment:
I think this should return a `std::string` rather than `const char*`. When the `reference_wrapper` goes out of scope at the end of this function, we can't be sure if the `const char*` returned from `git_reference_shorthand` is valid or not without checking the `libgit2` code or documentation. If we return a `string` then we're taking a copy and this is guaranteed to work.